conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/DefaultStaticFactoryMethodGenerator.java

```diff
@@ -183,6 +238,21 @@ private CodeBlock constructDeserializer(Optional<Type> type, TypeName className)
                 : CodeBlock.of("emptyBodyDeserializer()");
     }
 
+    private CodeBlock constructDeserializerWithExceptions(Optional<Type> type, TypeName className) {
+        CodeBlock.Builder builder = CodeBlock.builder();
+        if (type.isEmpty()) {
+            builder.add("emptyBodyDeserializer(");
+        } else if (returnTypes.isBinary(className)) {
+            builder.add("inputStreamDeserializer(");
+        } else if (returnTypes.isOptionalBinary(className)) {
+            builder.add("optionalInputStreamDeserializer(");
+        } else {
+            builder.add("deserializer(");
+        }
```

Comment:
Is there a way we could make this more resilient to adding some other cases in the future? e.g. easier to discover we need to update this code location?